ros2param/ros2param/verb/dump.py

```diff
@@ -0,0 +1,146 @@
+            value = pvalue.byte_array_value
+        elif pvalue.type == ParameterType.PARAMETER_BOOL_ARRAY:
+            value = pvalue.bool_array_value
+        elif pvalue.type == ParameterType.PARAMETER_INTEGER_ARRAY:
+            value = pvalue.integer_array_value
+        elif pvalue.type == ParameterType.PARAMETER_DOUBLE_ARRAY:
+            value = pvalue.double_array_value
+        elif pvalue.type == ParameterType.PARAMETER_STRING_ARRAY:
+            value = pvalue.string_array_value
+        elif pvalue.type == ParameterType.PARAMETER_NOT_SET:
+            value = None
+        else:
+            value = None
+
+        return value
+
+    def insert_dict(self, dict, key, value):
+        split = key.split(PARAMETER_SEPARATOR_STRING, 1)
+        if len(split) > 1:
+            if not split[0] in dict:
+                dict[split[0]] = {}
+            self.insert_dict(dict[split[0]], split[1], value)
+        else:
+            dict[key] = value
+
+    def main(self, *, args):  # noqa: D102
+
+        with NodeStrategy(args) as node:
+            node_names = get_node_names(
+                node=node, include_hidden_nodes=args.include_hidden_nodes)
+
+        absolute_node_name = get_absolute_node_name(args.node_name)
+        node_name = parse_node_name(absolute_node_name)
+        if absolute_node_name:
+            if absolute_node_name not in [n.full_name for n in node_names]:
+                return 'Node not found'
+
+        if not os.path.isdir(args.file_path):
+            return 'Invalid output directory'
+
+        with DirectNode(args) as node:
+            # create client
+            service_name = '{absolute_node_name}/list_parameters'.format_map(locals())
+            client = node.create_client(ListParameters, service_name)
+
+            client.wait_for_service()
+
+            if not client.service_is_ready():
+                return 'Something went wrong'
+
+            request = ListParameters.Request()
+            future = client.call_async(request)
+
+            # wait for response
+            rclpy.spin_until_future_complete(node, future)
+
+            yaml_output = {node_name.name: {'ros__parameters': {}}}
+
+            # retrieve values
+            if future.result() is not None:
+                response = future.result()
+                for param_name in sorted(response.result.names):
+                    pval = self.get_parameter_value(node, absolute_node_name, param_name)
+                    self.insert_dict(
+                        yaml_output[node_name.name]['ros__parameters'], param_name, pval)
+            else:
+                e = future.exception()
+                print(
+                    'Exception while calling service of node '
+                    "'{node_name.full_name}': {e}".format_map(locals()),
+                    file=sys.stderr)
+
+            print('Saving to: ', os.path.join(args.file_path, node_name.name + ".yaml"))
```

Comment:
There are a few places that assume a fully qualified node name is unique, so making the output file name include the namespace seems appropriate. Recommend replacing `node_name.name` with `absolute_node_name.replace('/', '__')`

Reply:
The output file name and only that ? Not sure how it works with namespaces now that the node name is the first entry in the yaml file...  
Also should I filter for leading `' / '` in `absolute_node_name` ?

Reply:
> The output file name and only that ?

Yeah, I was just thinking it would make this case more convenient. It looks like the second call would overwrite the first if only using `node_name.name` for the file name.

```
ros2 param dump /left/camera --output-dir .
ros2 param dump /right/camera --output-dir .
```

> Also should I filter for leading ' / ' in absolute_node_name ?

Sure. The absolute name always begins with `/`, so that should be easy to remove.

Reply:
Done e41232cf37661afcd077a66cb0d891715a3fa9e1